docs/devel/offline-virtual-machine.md

```diff
@@ -0,0 +1,234 @@
+kubectl describe offlinevirtualmachine myvm
+
+# Look at the now created VirtualMachine status and associated events:
+kubectl describe virtualmachine myvm
+
+# Stop an OfflineVirtualMachine:
+kubectl patch offlinevirtualmachine myvm -p \
+    '{"spec":{"running":"false"}}'
+
+# Implicit cascade delete (first deletes the vm and then the ovm)
+kubectl delete offlinevirtualmachine myvm
+
+# Explicit cascade delete (first deletes the vm and then the ovm)
+kubectl delete offlinevirtualmachine myvm --cascade=true
+
+# Orphan delete (The running vm is only detached, not deleted)
+# Recreating the ovm would lead to the adoption of the vm
+kubectl delete offlinevirtualmachine myvm --cascade=false
+```
+
+### The REST API
+
+The kubectl is a handy tool that provides handy access to cluster, when you sit
+at the console. But, when you are writting an external application that
+needs to access the cluster programaticaly, it is better to have a API endpoint.
+Thats where the Kubernetes REST API endpoint comes right in. Kubernetes provides
+for its users the native REST API, which is easily extendable and in one place.
+
+The OfflineVirtualMachine object is a CRD, which implies that Kubernetes
+provides the API automatically. The API is located at the path
+
+```text
+<your-api-server-adress>/apis/kubevirt.io/v1alpha/offlinevirtualmachine/
+```
+
+and you can do typical REST manipulation, you would expect. All CRUD is
+supported, as shown in following example.
+
+```text
+POST /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine
+GET /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine
+GET /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+DELETE /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+PUT /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+PATCH /apis/kubevirt.io/v1alpha1/namespaces/{namespace}/offlinevirtualmachine/{name}
+```
+
+By **POST** you can store new object in the etcd. With **GET** you either
+get the list of all OfflineVirtualMachines or get the concrete one. **DELETE**
+removes the object from etcd and all its resources. If you want to update the
+existing OfflineVirtualMachine object use **PUT** and if you want to change
+an item inside the object use **PATCH**.
+More details on the API are in the [documentation](https://kubevirt-incubator.github.io/api-reference/content/operations.html).
+
+To data format used when communicating with the API is the JSON. The format is
+set up the ussual way by setting the Content-Type header to 'application/json'.
+The 'application/yaml' can also be used.
+
+## OfflineVirtualMachine object content
+
+Now its time to discuss the content of the OfflineVirtualMachine object.
+The object is defined in the same way as any other Kubernetes object. You can
+use YAML of JSON format to specify the content. The example structure is bellow:
+
+```yaml
```

Comment:
Are we going to add here the definition what are possible params with acceptable values?

Reply:
Yes, even though it is cumbersome. It is possible to connect to it. You can do it by proxying from kubectl or with the access token, connect to it directly. More info is written [here](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-cluster-api)

Reply:
I am not asking how to do it but to define it here or give a link.